python/pyarrow/tests/test_pandas.py

```diff
@@ -2781,6 +2781,21 @@ def test_dictionary_with_pandas():
     tm.assert_series_equal(pd.Series(pandas2), pd.Series(ex_pandas2))
 
 
+def test_variable_dictionary_with_pandas():
+    a1 = pa.DictionaryArray.from_arrays([0, 1, 2], ['a', 'b', 'c'])
+    a2 = pa.DictionaryArray.from_arrays([0, 1], ['a', 'c'])
```

Comment:
You might want to add some dictionary values in each that are not found in the other

Reply:
'b' is already in the first but not in the second (but can add also to the second an item not in the first).

Reply:
Right, that's what I meant =)